src/alcotest/core.ml

```diff
@@ -270,13 +302,14 @@ module Make (M : Monad.S) = struct
 
   let bold_s fmt = color `Bold fmt
 
-  let pp_error ~verbose ~doc_of_path ~output_file ppf (path, error) =
+  let pp_error ~verbose ~doc_of_path ~output_file ~max_log_lines ppf
+      (path, error) =
     let logs =
       let filename = output_file path in
       if verbose || not (Sys.file_exists filename) then Fmt.strf "%s\n" error
       else
         let file = open_in filename in
-        let output = string_of_channel file in
+        let output = read_tail ~max_lines:max_log_lines file in
```

Comment:
instead of allocating an intermediate buffer, `read_tail` could just returns a list of `strings` and `Fmt.strf` could iter over it. (note this was already the case before that patch, but we never noticed ;p)

Reply:
Do you think it matters? The current, naive implementation of tail involves loading the whole file in memory anyway. This won't prevent a memory blowup, which may be a concern with really large logs. Tailing a file from the end seems a lot trickier. Perhaps we could file this as a separate issue. At least I'll leave a comment.